### PR TITLE
Added tests for Deprovision and Unbind.

### PR DESCRIPTION
**Pull request by SamiSousa**
Also added test for whitelist entries.
Added Ping to the Provision step.

### CODE REVIEW

File: tests/broker_test.go [hunk 1]
```diff
@@ -53,12 +59,65 @@ func TestBrokerLogic(t *testing.T){
...
 	// we want an error here
 	if errProvisionImproper == nil {
-		t.Errorf("Error on Provision with invalid token: no error returned")
+		t.Errorf("Error on Provision with invalid token: no error returned\n")
+	}
+
+	// Instance doesn't exist
+	_, errProvisionFake := businessLogic.Provision(
+		&osb.ProvisionRequest{
+			InstanceID:	"not-an-instance",
```
Comment (danmcp) [inline tests/broker_test.go]: Formatting is a mess here.  Has this run through go format?  Ideally a format check should be part of travis.

File: tests/whitelist_test.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+			uuids[dataverse.ServiceID] = 1
+		}
+
+		if _, plan_present := uuids[dataverse.PlanID]; plan_present {
+			t.Errorf("Error in whitelist: Dataverse Service %s does not have unique PlanID\n", dataverse.ID)
+		} else {
+			uuids[dataverse.PlanID] = 1
+		}
+
+		// Existance
```
Comment (danmcp) [inline tests/whitelist_test.go]: Spelling

File: tests/whitelist_test.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	"strings"
+
+	logic "github.com/dataverse-broker/dataverse-broker/pkg/broker"
+)
+
+/* Check all whitelisted services for validity:
+	- Completeness: Has all required info for a dataverse service
+		(server name, identifier, Name, etc)
+	- Uniqueness: Unique service ids and plan ids, etc
+	- Existance: Pings the server to see if the dataverse exists/is live
```
Comment (danmcp) [inline tests/whitelist_test.go]: Spelling

**Comment by danmcp** [comment]
@SamiSousa Please squash these commits.

**Comment by SamiSousa** [comment]
Fixed typos. Added pre-commit hook for automatic go formatting on commits. Formatted all code in pkg/broker and tests.
We'll squash when we merge this into master.

**Comment by danmcp** [comment]
@SamiSousa Little confused about your squash statement.  Why wouldn't you squash it in the PR?  I assume you will be merging the PR to get it into master.

**Comment by danmcp** [comment]
@SamiSousa Also to be clear, it doesn't have to squashed into 1 commit because you have multiple authors.  But at least remove in intermediate state commits.

**Comment by SamiSousa** [comment]
Actually the changes are already merged to master, I erroneously left them on my branch.

**Comment by SamiSousa** [comment]
Added a format check to Travis, and ran gofmt on all go files. IShould I also include a script to automatically format all go code within the repo?

File: pkg/broker/types.go [hunk 4]
```diff
@@ -19,72 +19,69 @@ type BusinessLogic struct {
...
-	Type string `json:"type"`
+type DatafileChecksum struct {
+	Type  string `json:"type"`
 	Value string `json:"value"`
 }
 
 // type for JSON portion describing a dataverse on Server
-type DataverseDescription struct{
+type DataverseDescription struct {
 	// Wow, capitalization matters for structs in go?
```
Comment (ckl2150) [inline pkg/broker/types.go]: Can probably take this out now heh

Comment (SamiSousa) [reply pkg/broker/types.go]: Nice catch. I'll clean up my comments

**Comment by ckl2150** [comment]
Other than inclusion of gofmt, looks like a lot of quality-of-life changes (formatting, etc). Looks good to me.

**Comment by SamiSousa** [comment]
Cleaned up comments

**Pull request merged**